packages/frontend/src/components/RootBoundary.tsx

```diff
@@ -0,0 +1,24 @@
+import { isRouteErrorResponse, useRouteError } from 'react-router-dom';
+import { AccessDenied } from '../pages/AccessDenied';
+import { NotFound } from '../pages/NotFound';
+
+// TODO add more cases here
+export function RootBoundary() {
+  const error = useRouteError();
+
+  if (isRouteErrorResponse(error)) {
+    if (error.status === 404) {
+      return <NotFound />;
+    }
+
+    if (error.status === 401) {
```

Comment:
@saneDG Could you try to throw error status 401 in `action` or `loader` in routes.tsx and see if can point at <AccessDenied />?